Install add_extensions in flapjack setup

ptomato opened the pull request:
If there's an extension listed in the `add_extensions` config key, then we
want to install it at `flapjack setup` time. Otherwise, `flapjack build` will
immediately fail, giving a bad first experience.

Closes #15.

ptomato committed:
ext: Decode ext.git() and ext.flatpak() output

I can't think of a situation where we wouldn't want to decode these in
the default encoding.

ptomato committed:
ext: Small code cleanup

ptomato committed:
flapjack setup: Install add_extensions

Any extensions mentioned in the add_extensions config key should be
installed by 'flapjack setup'. Since they may be located in different
remotes from the main runtime, search all remotes for each extension.

Closes #15.

manuq commented on flapjack/ext.py:
```diff
@@ -40,9 +41,9 @@ def flatpak(command, *args, output=False, code=False):
+    cmdline = ['flatpak', command] + user_arg + list(args)
     if output:
-        return subprocess.check_output(cmdline)
+        return subprocess.check_output(cmdline, universal_newlines=True)
```
The `universal_newlines` parameter is adding a `\n` before `http://` if using the current example config. This makes `flapjack setup` fail using the example INI file.

```
subprocess.CalledProcessError: Command '['flatpak', 'remote-add', '--if-not-exists', '--from', 'flapjack-source', '\nhttp://endlessm.github.io/eos-knowledge-lib/eos-sdk-nightly.flatpakrepo']' returned non-zero exit status 1
```

We should remove the newline after the equal sign here:
```
# sdk_repo_name = eos-sdk-nightly
sdk_repo_definition =
    http://endlessm.github.io/eos-knowledge-lib/eos-sdk-nightly.flatpakrepo
```

ptomato replied:
Good catch. However I think it's better to strip whitespace from config keys, in order to avoid this problem in the future. I've added a commit for that.

manuq replied:
@ptomato good point. Also, I just noticed this is not an issue introduced by these changes, The universal_whitespace parameter is for output only. Testing your fix to strip whitespace. If that works I'll merge the PR.

manuq reviewed (changes requested):
See note on `\n`. I removed the newline in the INI file and after that, this brach works like a charm.

ptomato committed:
config: Strip config values by default

I didn't realize that the default config file getter preserves
whitespace. We should strip whitespace from config keys by default.
